### ISSUE TITLE
BUG: `---` disables spellchecking in .md file

### ISSUE THREAD

**Issue by adrienjoly** [adrienjoly/vscode-spellchecker#2]
```md
element point. ellement. je suis un elllement. yeah
élèment

---

element point. ellement. je suis un elllement. yeah
élèment
```

=>

<img width="482" alt="screen shot 2017-02-15 at 12 28 16" src="https://cloud.githubusercontent.com/assets/531781/22972554/612a8be2-f37a-11e6-9fb5-39547bb229b7.png">

**Comment by adrienjoly** [comment]
unable to reproduce, even on initial context (5.md file)

**Comment by adrienjoly** [comment]
Finally able to reproduce the bug on another file: http://pastebin.com/60kgvn2j

"Bonjouur" is not spotted as mispelled, even though spelling errors are spotted farther in the document:

<img width="445" alt="screen shot 2017-03-15 at 13 41 31" src="https://cloud.githubusercontent.com/assets/531781/23948836/26c2d9b6-0985-11e7-8511-92e717084939.png">

**Comment by adrienjoly** [comment]
partially fixed by [v1.2.11](https://github.com/adrienjoly/vscode-spellchecker/releases/tag/v1.2.11)

video of the fix: https://www.twitch.tv/videos/128833002 / https://www.youtube.com/watch?v=7j1rO4h9Vr0

**Comment by adrienjoly** [comment]
## canonical test file, to reveal the bug

```md
---   <- skipsError 
 
bonjouur

---

- (ex: id, href, src...).   <- skipsError
```

expected behavior: `bonjouur` should appear as misspelled.

observed behavior: `bonjouur` does not appear as misspelled.

## observations

- deleting the first or second `skipsError` line makes `bonjouur` appear as misspelled.

## fix

disable the line:

```js
text = text.replace( /---(.|\n)*\.\.\./g, ' ' ); // TODO: activate on yaml files only
```

from `doSpellCheck()`
